controller/controller.go

```diff
@@ -16,26 +21,144 @@ import (
 	"github.com/go-chi/chi"
 )
 
-// LnxEndpoint specifies the remote Lnx translate server used by
-// brave-core, and it can be set to a mock server during testing.
-var LnxEndpoint = os.Getenv("LNX_HOST")
-var LnxAPIKey = os.Getenv("LNX_API_KEY")
-var languagePath = "/get-languages"
-var translatePath = "/translate"
+var (
+	LnxEndpoint   *LnxEndpointConfiguration
+	LnxAPIKey     = os.Getenv("LNX_API_KEY")
+	languagePath  = "/get-languages"
+	translatePath = "/translate"
+)
+
+// LnxEndpointConfiguration describes a configuration of lingvanex endpoints, their supported
+// languages and weights.
+type LnxEndpointConfiguration struct {
+	// A list of endpoint URLs.
+	Endpoints []string
+	// A list of default endpoint weights.
+	DefaultWeights []float64
+	// A GoogleLanguageList containing source language descriptions and target language descriptions.
+	LanguagePairList language.GoogleLanguageList
+	// A nested map of endpoint weights for a language pair.
+	// The first key represents the source language, the second key represents the target language, the 
+	// third key represents the endpoint URL and the value the corresponding weight for that endpoint.
+	LanguagePairWeights map[string]map[string]map[string]float64
```

Comment:
minor: for me `LanguagePairWeights[tl][sl]` looks more like an array of pair (str, float64) than `map[string]float64`. It seems we don't actually use endpoint as a index.

Reply:
ah - yes you could definitely write it that way instead. i guess this way if one were to accidentally duplicate an endpoint in the list of endpoints then this would accomplish de-duplication and avoid overweighting?

in practice, shouldn't matter much.